PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx

```diff
@@ -0,0 +1,1111 @@
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi) {
+        mCasc = casc.mXi();
+      } else {
+        mCasc = casc.mOmega();
+      }
+      auto arrMassCharmBaryon = std::array{0., 0.};
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi) {
+        arrMassCharmBaryon = {massXiFromPDG, massPionFromPDG};
+      } else if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi) {
+        arrMassCharmBaryon = {massOmegaFromPDG, massPionFromPDG};
+      } else {
+        arrMassCharmBaryon = {massOmegaFromPDG, massKaonFromPDG};
+      }
+      double mCharmBaryon = RecoDecay::m(std::array{pVecCascAsD, pVecCharmBachelorAsD}, arrMassCharmBaryon);
+
+      // computing cosPA
+      double cpaV0 = casc.v0cosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaCharmBaryon = RecoDecay::cpa(pvCoord, coordVtxCharmBaryon, pVecCharmBaryon);
+      double cpaCasc = casc.casccosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaxyV0 = RecoDecay::cpaXY(pvCoord, vertexV0, pVecV0);
+      double cpaxyCharmBaryon = RecoDecay::cpaXY(pvCoord, coordVtxCharmBaryon, pVecCharmBaryon);
+      double cpaxyCasc = RecoDecay::cpaXY(pvCoord, vertexCasc, pVecCasc);
+
+      // computing decay length and ctau
+      double decLenCharmBaryon = RecoDecay::distance(pvCoord, coordVtxCharmBaryon);
+      double decLenCascade = RecoDecay::distance(coordVtxCharmBaryon, vertexCasc);
+      double decLenV0 = RecoDecay::distance(vertexCasc, vertexV0);
+
+      double phiCharmBaryon, thetaCharmBaryon;
+      getPointDirection(std::array{primaryVertex.getX(), primaryVertex.getY(), primaryVertex.getZ()}, coordVtxCharmBaryon, phiCharmBaryon, thetaCharmBaryon);
+      auto errorDecayLengthCharmBaryon = std::sqrt(getRotatedCovMatrixXX(primaryVertex.getCov(), phiCharmBaryon, thetaCharmBaryon) + getRotatedCovMatrixXX(covVtxCharmBaryon, phiCharmBaryon, thetaCharmBaryon));
+      auto errorDecayLengthXYCharmBaryon = std::sqrt(getRotatedCovMatrixXX(primaryVertex.getCov(), phiCharmBaryon, 0.) + getRotatedCovMatrixXX(covVtxCharmBaryon, phiCharmBaryon, 0.));
+
+      double ctOmegac = RecoDecay::ct(pVecCharmBaryon, decLenCharmBaryon, massOmegacFromPDG);
+      double ctXic = RecoDecay::ct(pVecCharmBaryon, decLenCharmBaryon, massXicFromPDG);
+      double ctCascade = 0.;
+      if constexpr (decayChannel == hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi) {
+        ctCascade = RecoDecay::ct(pVecCasc, decLenCascade, massXiFromPDG);
+      } else {
+        ctCascade = RecoDecay::ct(pVecCasc, decLenCascade, massOmegaFromPDG);
+      }
+      double ctV0 = RecoDecay::ct(pVecV0, decLenV0, massLambdaFromPDG);
+
+      // computing eta
+      double pseudorapCharmBaryon = RecoDecay::eta(pVecCharmBaryon);
+      double pseudorapCascade = RecoDecay::eta(pVecCasc);
+      double pseudorapV0 = RecoDecay::eta(pVecV0);
```

Comment:
All these variables are `float` in the output table, is it necessary to use `double` precision here instead?

Reply:
Okay, switched to float for all of them